### PR TITLE
[FIX,WIP] Setting inputs.ignore_exception

### LINKED ISSUE #2411: Update to 1.0.1-dev broke master
Seems like building docs fails for one workflow (`nipype/workflows/smri/ants/ANTSBuildTemplate.py`). 

Apparently, the `ignore_exception` input should be dropped in favor of the new class attribute in interfaces. Before 1.0.0, nipype would only show a warning, but now after the version bump to 1.0.1-dev it raises an error (#2403).

We need to fix a couple of things:
  - [ ] Make sure no legacy code remains setting `interface.inputs.ignore_exception`
  - [ ] Make sure interfaces no longer support `ignore_exception` and `terminal_output` as inputs. In particular, double check with the `Function` interface.

https://circleci.com/gh/nipy/nipype/3178
https://circleci.com/gh/nipy/nipype/3176
https://circleci.com/gh/nipy/nipype/3172

```
Exception occurred in traits notification handler.
Please check the log file for details.
Exception occurred in traits notification handler for object: 
AffineTransformList = <undefined>
function_str = def MakeListsOfTransformLists(warpTransformList, AffineTransformList):
    return list(map(list, list(zip(warpTransformList, AffineTransformList))))

ignore_exception = True
warpTransformList = <undefined>
, trait: ignore_exception, old value: False, new value: True
Traceback (most recent call last):
  File "/opt/conda/envs/neuro/lib/python3.6/site-packages/traits/trait_notifiers.py", line 519, in _dispatch_change_event
    self.dispatch( handler, *args )
  File "/opt/conda/envs/neuro/lib/python3.6/site-packages/traits/trait_notifiers.py", line 482, in dispatch
    handler( *args )
  File "/src/nipype/nipype/interfaces/base/specs.py", line 140, in _deprecated_warn
    raise TraitError(msg)
traits.trait_errors.TraitError: Input ignore_exception in interface Function is deprecated. Will be removed or raise an error as of release 1.0.0 

180126-21:20:31,898 traits ERROR:
	 Exception occurred in traits notification handler for object: 
AffineTransformList = <undefined>
function_str = def MakeListsOfTransformLists(warpTransformList, AffineTransformList):
    return list(map(list, list(zip(warpTransformList, AffineTransformList))))

ignore_exception = True
warpTransformList = <undefined>
, trait: ignore_exception, old value: False, new value: True
Traceback (most recent call last):
  File "/opt/conda/envs/neuro/lib/python3.6/site-packages/traits/trait_notifiers.py", line 519, in _dispatch_change_event
    self.dispatch( handler, *args )
  File "/opt/conda/envs/neuro/lib/python3.6/site-packages/traits/trait_notifiers.py", line 482, in dispatch
    handler( *args )
  File "/src/nipype/nipype/interfaces/base/specs.py", line 140, in _deprecated_warn
    raise TraitError(msg)

traits.trait_errors.TraitError: Input ignore_exception in interface Function is deprecated. Will be removed or raise an error as of release 1.0.0 
Exception occurred in traits notification handler for object: 
averageAffineTranform = <undefined>
function_str = def MakeTransformListWithGradientWarps(averageAffineTranform,
                                       gradientStepWarp):
    return [
        averageAffineTranform, gradientStepWarp, gradientStepWarp,
        gradientStepWarp, gradientStepWarp
    ]

gradientStepWarp = <undefined>
ignore_exception = True
, trait: ignore_exception, old value: False, new value: True
Traceback (most recent call last):
  File "/opt/conda/envs/neuro/lib/python3.6/site-packages/traits/trait_notifiers.py", line 519, in_dispatch_change_event
    self.dispatch( handler, *args )
  File "/opt/conda/envs/neuro/lib/python3.6/site-packages/traits/trait_notifiers.py", line 482, in dispatch
    handler( *args )
  File "/src/nipype/nipype/interfaces/base/specs.py", line 140, in _deprecated_warn
    raise TraitError(msg)
traits.trait_errors.TraitError: Input ignore_exception in interface Function is deprecated. Will be removed or raise an error as of release 1.0.0 

180126-21:20:31,924 traits ERROR:
	 Exception occurred in traits notification handler for object: 
averageAffineTranform = <undefined>
function_str = def MakeTransformListWithGradientWarps(averageAffineTranform,
                                       gradientStepWarp):
    return [
        averageAffineTranform, gradientStepWarp, gradientStepWarp,
        gradientStepWarp, gradientStepWarp
    ]

gradientStepWarp = <undefined>
ignore_exception = True
, trait: ignore_exception, old value: False, new value: True
Traceback (most recent call last):
  File "/opt/conda/envs/neuro/lib/python3.6/site-packages/traits/trait_notifiers.py", line 519, in _dispatch_change_event
    self.dispatch( handler, *args )
  File "/opt/conda/envs/neuro/lib/python3.6/site-packages/traits/trait_notifiers.py", line 482, in dispatch
    handler( *args )
  File "/src/nipype/nipype/interfaces/base/specs.py", line 140, in _deprecated_warn
    raise TraitError(msg)

traits.trait_errors.TraitError: Input ignore_exception in interface Function is deprecated. Will be removed or raise an error as of release 1.0.0 
```

### PR DESCRIPTION
**Pull request by oesteban**
Fixes #2411

### CODE REVIEW

File: doc/users/saving_workflows.rst [hunk 1]
```diff
@@ -73,27 +73,27 @@ This will create a file "outputtestsave.py" with the following content:
 	    return out
 
 	testfunc.inputs.function_str = getsource(testfunc_1)
-	testfunc.inputs.ignore_exception = False
+	testfunc.ignore_exception = False
```
Comment (effigies) [inline doc/users/saving_workflows.rst]: .interface?

Comment (oesteban) [reply doc/users/saving_workflows.rst]: Yup, and I think that the class attribute does not exist either.

**Comment by satra** [comment]
@oesteban - sorry i merged this - didn't see the WIP classification. i reverted it.

**Comment by satra** [comment]
@oesteban - could you please reopen this request. in terms of changes, `ignore_exception` is an input option in `baseinterfaceinputspec`. so the changes should be `nodename.interface.inputs.ignore_exceotion = ...`

**Comment by oesteban** [comment]
Yes, I'll fix the remaining issues now.

**Comment by oesteban** [comment]
Continued in #2414

**Pull request merged**